main.lua

```diff
@@ -4472,8 +4512,28 @@ do -- Extra Rooms
                     StageAPI.TransitionToExtraRoom(data.DoorGridData.LeadsToExtra, data.DoorGridData.ExitSlot)
                 elseif data.DoorGridData.LeadsToNormal then
                     transitionStarted = true
-                    StageAPI.TransitionFromExtraRoom(data.DoorGridData.LeadsToNormal, data.DoorGridData.ExitSlot)
+
+                    local leadsToAdjacentRoom = false
+                    local slot = data.DoorGridData.Slot
+
+                    if not StageAPI.InExtraRoom then
+                        local slotIndex = StageAPI.GetRoomIdxRelativeToSlot(level:GetCurrentRoomDesc(), slot)
+                        leadsToAdjacentRoom = slotIndex == data.DoorGridData.LeadsToNormal
+                    end
+
+                    if leadsToAdjacentRoom then
+                        game:StartRoomTransition(data.DoorGridData.LeadsToNormal, slot % 4, 0)
+                    else
+                        StageAPI.TransitionFromExtraRoom(data.DoorGridData.LeadsToNormal, data.DoorGridData.ExitSlot)
```

Comment:
I feel like rather than trying to be clever about detecting if the slot we're going to is adjacent and thereby using the vanilla transition, it would make more sense for that to be a setting on the door. Can you explain why you went for this approach?